plugwise/helper.py

```diff
@@ -1142,78 +1139,81 @@ def _schemas(self, location):
 
         return available, selected, schedule_temperature, last_used
 
-    def _last_used_schedule(self, loc_id, rule_ids):
+    def _last_used_schedule(self, loc_id, rule_ids) -> str | None:
         """Helper-function for smile.py: _device_data_climate().
         Determine the last-used schedule based on the location or the modified date.
         """
         # First, find last_used == selected
-        last_used = self._last_active.get(loc_id)
+        last_used: str | None = self._last_active.get(loc_id)
         if last_used is not None:
             return last_used
 
         # Alternatively, find last_used by finding the most recent modified_date
-        epoch = dt.datetime(1970, 1, 1, tzinfo=pytz.utc)
-        schemas = {}
+        epoch: dt.datetime = dt.datetime(1970, 1, 1, tzinfo=pytz.utc)
+        schemas: dict[str] | None = {}
 
         if not rule_ids:
             return  # pragma: no cover
```

Comment:
This check should be first in this method, guarding an early exit of the method when no rule IDs are requested